### PR TITLE
Show frequencies

### PR DESCRIPTION
**Pull request by akoryagin**
Add method `show_frequencies` that shows Fourier-spectrum of traces taken from specific component. Useful to look at before applying `bandpass_filter`.

### CODE REVIEW

File: seismiqb/crop_batch.py [hunk 1]
```diff
@@ -1463,3 +1464,68 @@ def show_roll(self, n=1, components=None, zoom_slice=None, displayed_name=None,
...
+        values in Hz while y-axis stands for amplitudes of specific frequencies.
+
+        For the analysis the method selects specific traces and item-indices. Traces are 1d slices taken along an axis.
+
+        Uses units of Nyquist frequency when depicting the spectrum. In this way achieves consistency with
+        action `bandpass_filter`.
+
+        Parameters
+        ----------
+        indices: int or tuple or list
```
Comment (HollowPrincess) [inline seismiqb/crop_batch.py:1480]:
```suggestion
        indices: int or sequence of int
```
Is more consistent with other docstrings.

Comment (HollowPrincess) [inline seismiqb/crop_batch.py:1480]: Please, add a space before colons in argument data types descriptions, like everywhere in docstrings.

File: seismiqb/crop_batch.py [hunk 2]
```diff
@@ -1463,3 +1464,68 @@ def show_roll(self, n=1, components=None, zoom_slice=None, displayed_name=None,
             **kwargs
         }
         return plot_image(data, **kwargs)
+
+    def show_frequencies(self, indices=(0, ), src='images', trace_indices=((0, 0), (-1, -1)), axis=2, d=None,
```
Comment (AlexeyKozhevin) [inline seismiqb/crop_batch.py:1468]: At the first line you convert integer `indices` to tuple, so you can define it as 0. It is prettier

File: seismiqb/crop_batch.py [hunk 3]
```diff
@@ -1463,3 +1464,68 @@ def show_roll(self, n=1, components=None, zoom_slice=None, displayed_name=None,
...
+
+            for trace_idx in trace_indices:
+                trace_idx_ = tuple(np.insert(np.array(trace_idx, dtype=np.object_), insert_index, slice(0, None)))
+                amplitudes = fft(data[trace_idx_])[1 : data.shape[axis]//2]
+                plot_data.append((frequencies, np.abs(amplitudes)))
+                plot_label.append(f'FIELD: {displayed_name}   IDX: {idx}   TRACE: {trace_idx}')
+
+        plot_params = {'title': f'Spectrum of {src}-component',
+                       'label': plot_label,
+                       'xlabel': 'Frequency, HZ',
```
Comment (AlexeyKozhevin) [inline seismiqb/crop_batch.py:1525]:
```suggestion
                       'xlabel': 'Frequency, Hz',
```

File: seismiqb/crop_batch.py [hunk 4]
```diff
@@ -1463,3 +1464,68 @@ def show_roll(self, n=1, components=None, zoom_slice=None, displayed_name=None,
...
+
+            data = self.get(self.indices[idx], src)
+            freqs = fftfreq(data.shape[axis], d)
+            frequencies = freqs[1 : data.shape[axis]//2]
+
+            for trace_idx in trace_indices:
+                trace_idx_ = tuple(np.insert(np.array(trace_idx, dtype=np.object_), insert_index, slice(0, None)))
+                amplitudes = fft(data[trace_idx_])[1 : data.shape[axis]//2]
+                plot_data.append((frequencies, np.abs(amplitudes)))
+                plot_label.append(f'FIELD: {displayed_name}   IDX: {idx}   TRACE: {trace_idx}')
```
Comment (AlexeyKozhevin) [inline seismiqb/crop_batch.py:1521]: Make indexing more clear to recover position in the geometry (make it in field coordinates).
`trace_idx` is indexing of the traces inside the crops in batch.

**Review by SergeyTsimfer — APPROVED** [review]

**Pull request merged**